lib/tether/interfaces.go

```diff
@@ -56,6 +56,23 @@ type Operations interface {
 	HandleUtilityExit(pid, exitCode int) bool
 }
 
+// System is a very preliminary interface that provides extensions with a means by which to
+// perform system style operations.
+// This should be designed with intent to provide sufficient abstraction that vic-init, tether
+// can be run by vcsim. Odds are that a significant portion of the methods on this interface
+// will be defined in vcsim and this interface will simply embed the vcsim interface.
+// There may well be overlap with shared.Sys that needs to be resolved
+//
+// For now its sole purpose is to allow extensions access to the specific portions of
+// BaseOperations that they actively use.
+type System interface {
```

Comment:
There is a System interface in lib/system, you likely can modify that for your use.

Reply:
@hmahmood , @hickeng wrote this commit to expose this object from the tether package to tether extensions to give me access to the pidMap/childReaper so I'll let him comment as necessary but I suspect that whatever is in `lib/system` isn't what I want here.

Reply:
@hmahmood yeah - I suspect there'll be some consolidation onto that eventually, but this is explicitly a subset of `Operations`. I've not removed these methods from `Operations` and used composition with this structure because I think this is all liable to change. As noted in the comment there's conceptual overlap with `shared.Sys` which is the `lib/system` that you mention. 

Basically I don't like the PR I submitted for @gigawhitlocks to expose LaunchUtility in a manner that's useful to him - it's layered on top of some primordial tech debt around extensions and how they interact with the environment. I want to revisit this but don't want to invest much time in polishing the current state.